uaclient/contract.py

```diff
@@ -309,9 +309,29 @@ def process_entitlements_delta(
     :param series_overrides: Boolean set True if series overrides should be
         applied to the new_access dict.
     """
+    from uaclient.entitlements import entitlements_enable_order
+
     delta_error = False
     unexpected_error = False
-    for name, new_entitlement in sorted(new_entitlements.items()):
+
+    # We need to sort our entitlements because some of them
+    # depend on other service to be enable first. Additionally,
+    # we must be careful because no all entitlements that are
+    # provided by the contract are supported by UA.
+    def sort_entitlements(key_name):
+        ent_order = entitlements_enable_order(cfg)
+        try:
+            ent_pos = ent_order.index(key_name)
+        except ValueError:
+            ent_pos = len(ent_order)
+
+        return ent_pos
+
+    sorted_entitlements = sorted(
+        new_entitlements.items(), key=lambda ent: sort_entitlements(ent[0])
+    )
+
+    for name, new_entitlement in sorted_entitlements:
```

Comment:
Would it work to do something like this?
```suggestion
    for name in entitlements_enable_order(cfg):
        try:
            new_entitlement = new_entitlements[name]
        except KeyError:
            continue
```

Reply:
I like that refacor, even though the recursion is not an issue as you mentioned, but I think this is easier to read than my solution.